### ISSUE TITLE
Makefile Bugs

### ISSUE THREAD

**Issue by ramirezfranciscof** [MALBECC/lio#113]
To accumulate bugs in the makefile:
- [x] cublasmath should depend on general_module and it does not.
- [x] Include the path to cuda stubs in the makefile (see issue #53).
- [x] Link libg2g and liblioamber to the corresponding path without LD_LIB_PATH
- [x] Add  -Wl,--no-undefined and link all libs.

(see below for clarifications)

**Comment by ramirezfranciscof** [comment]
Include the path to cuda stubs in the makefile.

**Comment by ramirezfranciscof** [comment]
Link libg2g and liblioamber to the corresponding path without LD_LIB_PATH. See [this](https://stackoverflow.com/questions/2726993/how-to-specify-preference-of-library-path) and [this](https://stackoverflow.com/questions/3840218/how-to-specify-the-library-version-to-use-at-link-time) or [this](https://stackoverflow.com/questions/8835108/how-to-specify-non-default-shared-library-path-in-gcc-linux-getting-error-whil) or [this](https://stackoverflow.com/questions/828053/how-do-you-link-to-a-specific-version-of-a-shared-library-in-gcc)

**Comment by ramirezfranciscof** [comment]
Changes:

-LFLAGS += -fPIC -shared
+LFLAGS += -fPIC -shared -Wl,--no-undefined

-  LIBS   += -llapack
+  LIBS   += -lblas -llapack

-  LIBS   += -lmkl_lapack95_lp64 -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core
+  LIBS   += -lmkl_blas95_lp64 -lmkl_lapack95_lp64
+  LIBS   += -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core

**Comment by ramirezfranciscof** [comment]
Clean commits with all warnings

**Comment by ramirezfranciscof** [comment]
I'm using this issue to dump all makefile "bugs" so that I can fix them all in one commit (It has happened a few time I made a correction only to find out the day after that there was another problem that I could have dealt with). I can fix all these once the OS-PR series is complete.

Each post is a different modification, which one do you want me to explain? I think the only one that is confusing is the last one, in which I mean that when activating all warnings (analytics=2 if not mistaken) the compilation should be as clean as possible (yes, this should require a major restructure of many parts of the code; probably should be a whole separate issue or even many PRs).

**Comment by fedepedron** [comment]
Good, in that case, shouldn't we include #112 here?

**Comment by ramirezfranciscof** [comment]
#53 is the second post (it is good to have the reference) and the problem with #112 is that I don't have any clue on what is wrong with that so I don't know how to fix it yet.

**Comment by fedepedron** [comment]
Just to add to this, g2g does not recompile when altering pointxc files or changing compilation options such as analytics
